src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -99,29 +111,77 @@ const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
   }
 }
 
-export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
-  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
-    <Container>
-      <Icon source={require("../../../../../../images/payment_request.png")} />
-      <TextContainer>
-        <PaymentRequest>Payment request</PaymentRequest>
-        <CostLabel>
-          {invoice.total}
-        </CostLabel>
-      </TextContainer>
-      <TextContainer>
-        <InvoiceStateButton state={invoice.state} />
-      </TextContainer>
-    </Container>
-  </TouchableHighlight>
-
-export default createFragmentContainer(
+interface State {
+  hasReceivedNotification: boolean
+  subscription: EmitterSubscription
+}
+
+export class InvoicePreview extends React.Component<Props, State> {
+  componentWillMount() {
+    const { conversationId, relay, invoice } = this.props
+    const { ARNotificationsManager } = NativeModules
+    const notificationsManagerEmitter = new NativeEventEmitter(ARNotificationsManager)
+    const paymentRequestPaidCallback = notification => {
+      if (notification.url === invoice.payment_url) {
+        // Optimistically update the UI, but also refetch.
+        this.setState({
+          hasReceivedNotification: true,
+        })
+        relay.refetch({ conversationId, invoiceId: invoice.lewitt_invoice_id }, null, null, { force: true })
+      }
+    }
+    const subscription = notificationsManagerEmitter.addListener("PaymentRequestPaid", paymentRequestPaidCallback)
+
+    this.setState({
+      subscription,
+    })
+  }
+
+  componentWillUnmount() {
+    if (this.state.subscription) {
+      this.state.subscription.remove()
+    }
+  }
+
+  render() {
+    const { invoice, onSelected } = this.props
+
+    return (
+      <TouchableHighlight onPress={invoice.state === "UNPAID" ? onSelected : null} underlayColor={colors["gray-light"]}>
+        <Container>
+          <Icon source={require("../../../../../../images/payment_request.png")} />
+          <TextContainer>
+            <PaymentRequest>Payment request</PaymentRequest>
+            <CostLabel>
+              {invoice.total}
+            </CostLabel>
+          </TextContainer>
+          <TextContainer>
+            <InvoiceStateButton invoiceState={invoice.state} isOptimisticUpdate={this.state.hasReceivedNotification} />
```

Comment:
Because this component is a button it catches the touch and thus stops it from bubbling up to the `TouchableHighlight` on like 150 but it also does nothing with the event itself and so nothing happens in the end.